### DISCUSSION TITLE
`widgetActionDropdown`: Give us your thoughts!

### DISCUSSION THREAD

**Discussion by GerardasB** [iTwin/appui#723] [Preview Features]
# Feature description

The widget title bar will contain a single drop-down button when the specified `threshold` is exceeded by the button count in the widget title bar. To enable the drop-down menu for all the widgets assign the `threshold` to `0`.

The basis for this request was to allow further widget action additions without reducing the widget area dedicated to widget tabs.

[Live demo](https://itwin.github.io/appui/storybook/?path=/story/previewfeatures-widgetactiondropdown--default)

> [!NOTE]  
> Initially released in 4.9.0.

![Screen Recording 2024-04-11 at 5 18 43 PM](https://github.com/iTwin/appui/assets/849817/d17414e7-c444-4ab9-9205-621ea7709ea8)

## Why is it a preview feature?

- The dropdown menu depending on configuration might contain a few options and the overflow might not be needed.
- When enabled, widget actions are hidden and additional interaction from the user is needed.
- We want to determine if the new behavior should be the default and only option, or rather if it should be an "opt-in" and require related APIs to be configurable.

## What then?

We would like the community to chime in on any issues they are facing when using this feature, and discuss how we can resolve these. Additionally, we're interested in identifying any specific APIs that could enhance this feature, additional options, alternations or functionalities to be considered, and hearing any other ideas or opinions on the subject.

Feel free to add comments on its usefulness and how this feature should be officially supported!

**Comment by DanAbney** [comment]
What do we know about the widget area dedicated to the tabs? IE: do we know how many commands we should limit to before overflowing (switching) to the menu?

**Reply by FlyersPh9** [reply]
Currently the limit can be set by the development team, so it's fully customizable.  We do not know what limit would be the best default.

**Reply by DanAbney** [reply]
No worries ... did someone request this? Maybe they have some thoughts on what the guidelines could be, as a start at least.

**Reply by FlyersPh9** [reply]
It was requested by development teams to add additional items in the top right area.  This is our own solution to how we can handle so many items in this area.
